sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBufferCatalog.scala

```diff
@@ -232,6 +239,8 @@ object RapidsBufferCatalog extends Logging with Arm {
       gdsStorage.close()
       gdsStorage = null
     }
+
+    singleton.removeAllBuffersOnShutdown()
```

Comment:
Wondering if this will just move the exception somewhere else.  For example, could we now get "unknown buffer ID" if an outstanding buffer gets unregistered after catalog shutdown?

It feels like we're patching a symptom rather than fixing the root cause of why all the buffers didn't get unregistered for the first context.